### PR TITLE
Default result assertions fix 

### PR DESCRIPTION
**Pull request by JordanllHarper**
Implemented a fix which pasts the tests.
Using exception or null discarded inner exception value, so manually mapped to retain that value.
Let me know what I'm needing to do to improve the code
______________________________________
I confirm that I have read the [Contributor Agreements v1.0](https://github.com/robstoll/atrium/blob/main/.github/Contributor%20Agreements%20v1.0.txt), agree to be bound on them and confirm that my contribution is compliant.

### CODE REVIEW

File: apis/fluent-en_GB/extensions/atrium-api-fluent-en_GB-kotlin_1_3/src/commonTest/kotlin/ch/tutteli/atrium/api/fluent/en_GB/kotlin_1_3/samples/ResultExpectationSamples.kt [hunk 1]
```diff
@@ -57,69 +58,85 @@ class ResultExpectationSamples {
...
+        val failure = Result.failure<Int>(IllegalArgumentException(message))
+
+
+
+        expect(failure) {
+            toBeAFailure<IllegalArgumentException>()
+        }
+
+
+    }
```
Comment (robstoll) [inline apis/fluent-en_GB/extensions/atrium-api-fluent-en_GB-kotlin_1_3/src/commonTest/kotlin/ch/tutteli/atrium/api/fluent/en_GB/kotlin_1_3/samples/ResultExpectationSamples.kt:76]: Please remove this one again, suffices if we have it in ResultExpecationsSpec, here we only want samples and not regression tests. Also, the samples below already failed due to the bug, so we do not even need an additional regression here.

File: misc/atrium-specs/src/commonMain/kotlin/ch/tutteli/atrium/specs/integration/ResultExpectationsSpec.kt [hunk 2]
```diff
@@ -54,73 +54,73 @@ abstract class ResultExpectationsSpec(
...
-                    }.toThrow<AssertionError> {
-                        messageToContain("value: $isNotSuccessDescr")
-                        if (hasExtraHint) messageToContain("$toEqualDescr: 1")
-                    }
-                }
-            }
-
-            //TODO 0.20.0 activate once we have the workaround for #1234 implemented
+//
+//    fun describeFun(vararg pairs: SpecPair<*>, body: Suite.() -> Unit) =
```
Comment (robstoll) [inline misc/atrium-specs/src/commonMain/kotlin/ch/tutteli/atrium/specs/integration/ResultExpectationsSpec.kt:58]: I guess that was accidental, you should not comment out the tests

**Review by robstoll — CHANGES_REQUESTED** [review]

**Comment by robstoll** [comment]
@JordanllHarper make sure to rebase on main since you have conflicts

File: apis/fluent-en_GB/extensions/atrium-api-fluent-en_GB-kotlin_1_3/src/commonTest/kotlin/ch/tutteli/atrium/api/fluent/en_GB/kotlin_1_3/samples/ResultExpectationSamples.kt (hunk 1 above)
Comment (JordanllHarper) [reply apis/fluent-en_GB/extensions/atrium-api-fluent-en_GB-kotlin_1_3/src/commonTest/kotlin/ch/tutteli/atrium/api/fluent/en_GB/kotlin_1_3/samples/ResultExpectationSamples.kt:76]: Do you want me to comment out all the tests to the bottom or just the test?

**Review by robstoll — COMMENTED** [review]
btw. no need for a new PR, you can keep this one open. we are almost good 😊

**Comment by JordanllHarper** [comment]
If it's okay can we continue on the new one. As you can see, I was having some issues with git last night 😅.

**Comment by robstoll** [comment]
sure

**Pull request closed without merging**